ee/hogai/graph/graph.py

```diff
@@ -59,10 +59,10 @@ def add_node(self, node: AssistantNodeName, action: RunnableLike):
         self._graph.add_node(node, action)
         return self
 
-    def compile(self, checkpointer: DjangoCheckpointer | None = None):
+    def compile(self, checkpointer: DjangoCheckpointer | None | Literal[False] = None):
         if not self._has_start_node:
             raise ValueError("Start node not added to the graph")
-        return self._graph.compile(checkpointer=checkpointer or global_checkpointer)
+        return self._graph.compile(checkpointer=checkpointer if checkpointer is not None else global_checkpointer)
```

Comment:
Would it add value to have a comment here? As looks like we pass checkpointer even if it's false, so it's not obvious.

Reply:
For sure, added a comment to explain